client/state/sites/test/reducer.js

```diff
@@ -412,7 +412,7 @@ describe( 'reducer', () => {
 				}
 			} );
 
-			expect( state ).to.equal( original );
+			expect( state ).to.eql( original );
```

Comment:
That we needed to change the tests here implies that we're returning a new state reference in the reducer where we shouldn't be (no changes occurred). I think in your reducer's `switch` where you're breaking upon finding that there are no differences, we should `return memo;` instead to bypass the rest of the `reduce` and ensure that we only create a copy of the state object (return a new reference) if we're certain that a change has actually occurred.